litellm/llms/prompt_templates/factory.py

```diff
@@ -1016,7 +1014,13 @@ def anthropic_messages_pt(messages: list):
             msg_i += 1
 
         if user_content:
-            new_messages.append({"role": "user", "content": user_content})
+            if merge_with_previous:
+                new_messages[-1]["content"].extend(user_content)
```

Comment:
this assumes new_messages will always contain "content" if 'merge_with_previous' is true 

however -> merge_with_previous seems to be set if msg_i > 0, not if new_messages contains at least one item of content